runtime/compiler/runtime/SymbolValidationManager.hpp

```diff
@@ -848,6 +849,37 @@ class SymbolValidationManager
    typedef TR::typed_allocator<J9ClassLoader*, TR::Region&> LoaderAllocator;
    typedef std::vector<J9ClassLoader*, LoaderAllocator> LoaderVector;
    LoaderVector _loadersOkForWellKnownClasses;
+
+   // Remember which classes have already been found in which beholders'
+   // constant pools, regardless of CP index. If a class C has already been
+   // found in the constant pool of class D (producing a ClassFromCP record),
+   // then ClassByName(C, D) is redundant.
+   struct ClassFromAnyCPIndex
+      {
+      TR_OpaqueClassBlock *clazz;
+      TR_OpaqueClassBlock *beholder;
+
+      ClassFromAnyCPIndex(TR_OpaqueClassBlock *clazz, TR_OpaqueClassBlock *beholder)
+         : clazz(clazz), beholder(beholder) { }
+      };
+
+   struct LessClassFromAnyCPIndex
+      {
+      bool operator()(const ClassFromAnyCPIndex &a, const ClassFromAnyCPIndex &b) const
+         {
+         std::less<TR_OpaqueClassBlock*> lt;
```

Comment:
Curiosity question: how come you didn't use `LexicalOrder` here?

Reply:
I wanted to keep `LexicalOrder` file-local in `SymbolValidationManager.cpp`. It could be useful more generally, but we'd have to figure out where to put it—certainly not in `SymbolValidationManager.hpp`. I suppose we could also move the implementation of `LessClassFromAnyCPIndex::operator()` into the cpp file

Reply:
I see, I think it's fine as is for now; if ever there's a need for another "less" operator, we can then move the `LexicalOrder` class somewhere more appropriate.